src/view/event-edit-view.js

```diff
@@ -138,40 +144,70 @@ function createFormEditTemplate(points, offers, checkedOffers, destinations) {
   );
 }
 
-export default class EventEditView extends AbstractView {
-  #point = null;
+export default class EventEditView extends AbstractStatefulView {
   #offers = null;
   #checkedOffers = null;
   #destinations = null;
+  #destinationsAll = null;
   #handleFormSubmit = null;
   #handleEditClick = null;
 
-  constructor({point, offers, checkedOffers, destinations, onFormSubmit,onEditClick}) {
+  constructor({point, offers, checkedOffers, destinations, destinationsAll, onFormSubmit,onEditClick}) {
     super();
-    this.#point = point;
     this.#offers = offers;
     this.#checkedOffers = checkedOffers;
     this.#destinations = destinations;
+    this.#destinationsAll = destinationsAll;
     this.#handleFormSubmit = onFormSubmit;
     this.#handleEditClick = onEditClick;
 
-    this.element.querySelector('.event__rollup-btn')
-      .addEventListener('click', this.#editClickHandler);
-
-    this.element.addEventListener('submit', this.#formSubmitHandler);
+    this._setState(EventEditView.parsePointToState({point}));
+    this._restoreHandlers();
   }
 
   get template() {
-    return createFormEditTemplate(this.#point, this.#offers, this.#checkedOffers, this.#destinations);
+    return createFormEditTemplate(this._state, this.#offers, this.#checkedOffers, this.#destinations,this.#destinationsAll);
   }
 
-  #formSubmitHandler = (evt) => {
-    evt.preventDefault();
-    this.#handleFormSubmit(this.#point);
+  _restoreHandlers = () => {
+    this.element.querySelector('.event__rollup-btn').addEventListener('click', this.#editClickHandler);
+    this.element.querySelectorAll('.event__type-input').forEach((element) => element.addEventListener('click', this.#changeTypeHandler));
+    this.element.querySelector('.event__input--destination').addEventListener('click', this.#changeDestinationHandler);
+    this.element.addEventListener('submit', this.#formSubmitHandler);
   };
 
+
   #editClickHandler = (evt) => {
     evt.preventDefault();
     this.#handleEditClick();
+    console.log(this._state);
+  };
+
+  #changeTypeHandler = (evt) => {
+    evt.preventDefault();
+    this.updateElement({
+      type: `${evt.target.value}`,
+      offers: [],
+    });
+    console.log(this._state);
+  };
+
+  #changeDestinationHandler = (evt) => {
+    const selectedDestination = this.#destinationsAll.find((destination) => destination.name === `${evt.target. value}`);
+    const selectedDestinationId = (selectedDestination) ? selectedDestination.id : null;
```

Comment:
тут скобки не обязательно писать
```suggestion
    const selectedDestinationId = selectedDestination ? selectedDestination.id : null;
```